include/envoy/tracing/trace_context.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/common/pure.h"
+
+#include "absl/strings/string_view.h"
+#include "absl/types/optional.h"
+
+namespace Envoy {
+namespace Tracing {
+
+/**
+ * Protocol-independent abstraction for traceable stream. It hides the differences between different
+ * protocol requests and provides Tracer Driver with common methods for obtaining and setting the
+ * Tracing context.
+ */
+class TraceContext {
+public:
+  virtual ~TraceContext() = default;
+
+  /**
+   * Get tracing context value by key.
+   *
+   * @param key The context key of string view type. The context key should be a string view of a
+   * const lowercase string.
+   * @return The optional context value of string_view type.
+   */
+  virtual absl::optional<absl::string_view> getTraceContext(const absl::string_view key) const PURE;
```

Comment:
nit: no need top-level const in declarations.

```suggestion
  virtual absl::optional<absl::string_view> getTraceContext(absl::string_view key) const PURE;
```